libs/utils/utils.js

```diff
@@ -690,7 +690,8 @@ async function decorateIcons(area, config) {
   if (icons.length === 0) return;
   const { miloLibs, codeRoot } = config;
   const base = miloLibs || codeRoot;
-  await new Promise((resolve) => { loadStyle(`${base}/features/icons/icons.css`, resolve); });
+  loadStyle(`${base}/features/icons/icons.css`);
+  loadLink(`${base}/img/icons/icons.svg`, { rel: 'preload', as: 'fetch', crossorigin: 'anonymous' });
   const { default: loadIcons } = await import('../features/icons/icons.js');
```

Comment:
Since you don't wait for the css to finish loading at all, is there anything that stops `icons.js` from adding the icons to the page, an undecorated image showing up & only afterwards you add the CSS leading to CLS?

Reply:
The function name `loadStyle` is confusing. It just adds the CSS file to the HTML head.

Reply:
True, that's why you would need to promisify it, so that the callback within loadStyle/loadLink resolves the promise once the `onload` callback is called.

Otherwise, it's getting unpredictable when the CSS is ready

Reply:
The milo [loadBlock](https://github.com/adobecom/milo/blob/stage/libs/utils/utils.js#L454-L474) method loads the CSS and JS in parallel. However, it hides the section until everything is loaded.

Thanks @overmyheadandbody for finding those two critical lines:
https://github.com/adobecom/milo/blob/stage/libs/utils/utils.js#L766
https://github.com/adobecom/milo/blob/stage/libs/utils/utils.js#L1121

Reply:
I don't get what you were trying to prove. The file `icons.css` is only 22 lines and it's for tooltip styles, which won't affect CLS at all.